dsgrid/config/dataset_config.py

```diff
@@ -209,6 +222,10 @@ class DatasetConfigModel(DSGBaseModel):
     )
     # TODO: This must be validated against the project's dimension records for data_source
     # TODO: This must also be validated against the project_config
+    dataset_type_metadata: Optional[Any] = Field(
+        title="dataset_type_metadata",
+        description="Additional metadata to include related to dataset_type",
+    )
```

Comment:
So instead of overloading `dataset_type` I'd suggest having:

```
qualifiers: List[DatasetQualifiers]
```

and either

```
qualifier_data: List[Union(GrowthRateModel)]
```

or just

```
growth_rate_data: Optional[GrowthRateModel]
```

In either case, pydantic would need to enforce the presence of GrowthRateModel if qualifiers includes "growth_rate".

Reply:
I don't think this should be a list as we only want one type of data in a given dataset, right? E.g., consumption data would never be stored in the same dataframe as growth rate data

Reply:
I'm okay assuming there will be exactly one qualifier for now.

Should the type be Optional[GrowthRateModel] for now?